Automattic/jetpack
# JS Components: Convert ProductOffer component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `ProductOffer` component to TypeScript.
* It also converts the `ProductOffer` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `ProductOffer` component work as expected
* Check the consumers of `ProductOffer` and related components and confirm that they work as expected
* Check for any regression



https://user-images.githubusercontent.com/18226415/181243146-c92042a2-e053-4364-9a6e-79d216e0be82.mov



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314263990
